tracing-subscriber/Cargo.toml

```diff
@@ -32,6 +32,7 @@ fmt = ["registry", "std"]
 ansi = ["fmt", "nu-ansi-term"]
 registry = ["sharded-slab", "thread_local", "std"]
 json = ["tracing-serde", "serde", "serde_json"]
+wasm-bindgen = ["web-time"]
```

Comment:
> is this necessary as a feature flag, or can we just make the dependency always be used if `fmt` is enabled and we're on a WASM target?

We can.

> Are there reasons that a user might want to compile for a WASM target without WASI, but still disable this feature?

Yes, they might be using the `wasm32-unknown-unknown` target but not targeting the Web. Spans won't be working for them but they might still be using `tracing-subscriber` (as were Wasm on Web users until now). Though this is probably niche.

> Also, this feature flag should probably not introduce the additional dependency unless the `fmt` feature is also enabled.

I'm not aware of a way to do this. But if it's decided to enable it by default anyway we could just add it to the `fmt` crate feature.

---

I know that there are no-std Wasm users out there that can't rely on `wasm-bindgen`, but I don't know if they are using `tracing-subscriber` or not. It's also possible that a new Web backend gets popular, besides `wasm-bindgen`, in which case it would be useful to have separate crate features, though I'm not aware of any current development in that area.

Let me know how you want to proceed.